app/src/main/java/org/schabi/newpipe/download/DeleteManager.java

```diff
@@ -0,0 +1,178 @@
+
+        if (mPendingMap.size() == 1) {
+            showUndoDeleteSnackbar(mission);
+        }
+    }
+
+    public void setDownloadManager(@NonNull DownloadManager downloadManager) {
+        mDownloadManager = downloadManager;
+
+        if (mPendingMap.size() < 1) {
+            //nothing to do
+            return;
+        }
+
+        showUndoDeleteSnackbar();
+    }
+
+    public void restoreState(@Nullable Bundle savedInstanceState) {
+        if (savedInstanceState == null) {
+            // nothing to do
+            return;
+        }
+
+        List<String> list = savedInstanceState.getStringArrayList(KEY_STATE);
+        if (list != null) {
+            mPendingMap.addAll(list);
+        }
+    }
+
+    public void saveState(@Nullable Bundle outState) {
+        if (outState == null) {
+            // nothing to do
+            return;
+        }
+
+        for (Disposable disposable : mDisposableList) {
+            disposable.dispose();
+        }
+
+        outState.putStringArrayList(KEY_STATE, new ArrayList<>(mPendingMap));
+    }
+
+    private void showUndoDeleteSnackbar() {
+        if (mPendingMap.size() < 1) {
+            // nothing to do
+            return;
+        }
+
+        String url = mPendingMap.iterator().next();
+
+        for (int i = 0; i < mDownloadManager.getCount(); i++) {
+            DownloadMission mission = mDownloadManager.getMission(i);
+            if (url.equals(mission.url)) {
+                showUndoDeleteSnackbar(mission);
+                break;
+            }
+        }
+    }
+
+    private void showUndoDeleteSnackbar(@NonNull DownloadMission mission) {
+        final Snackbar snackbar = Snackbar.make(mView, mission.name, Snackbar.LENGTH_INDEFINITE);
+        final Disposable disposable = Observable.timer(3, TimeUnit.SECONDS)
+                .subscribeOn(AndroidSchedulers.mainThread())
+                .subscribe(l -> snackbar.dismiss());
+
+        mDisposableList.add(disposable);
+
+        snackbar.setAction(R.string.undo, v -> {
+            mPendingMap.remove(mission.url);
+            publishSubject.onNext(mission);
+            disposable.dispose();
+            snackbar.dismiss();
+        });
+
+        snackbar.addCallback(new BaseTransientBottomBar.BaseCallback<Snackbar>() {
+            @Override
+            public void onDismissed(Snackbar transientBottomBar, int event) {
+                if (!disposable.isDisposed()) {
+                    mPendingMap.remove(mission.url);
+                    Completable.fromAction(() -> deletePending(mission))
+                            .subscribeOn(Schedulers.io())
+                            .subscribe();
+                }
+                snackbar.removeCallback(this);
+                mDisposableList.remove(disposable);
+                showUndoDeleteSnackbar();
+            }
+        });
+
+        snackbar.show();
+    }
+
+    public void deletePending() {
+        if (mPendingMap.size() < 1) {
+            // nothing to do
```

Comment:
I think it's clear from the code that you want to abord the call here, so no need to commend it.

You could shorten it like this: `if(mPendingMap.size() < 1) return;`.